src/utils/utils_math.cpp

```diff
@@ -0,0 +1,24 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "utils_math.h"
+
+// Retrieves the position of the leftmost set bit.
+// The position of the bit is counted from 0
+// e.g. for 01000011110 the position equals 9.
+size_t getLeftmostSetBitPos(size_t num) {
+    // From C++20 countl_zero could be used for that.
+    size_t position = 0;
+    while (num > 0) {
+        num >>= 1;
+        position++;
+    }
+    position--;
```

Comment:
nit: It is better to use prefix form of increment/decrement operations in that case.

Reply:
Replaced this function with builtin.